server/src/main/java/org/elasticsearch/index/IndexMode.java

```diff
@@ -249,8 +249,18 @@ public void validateTimestampFieldMapping(boolean isDataStream, MappingLookup ma
         }
 
         @Override
-        public CompressedXContent getDefaultMapping() {
-            return DEFAULT_LOGS_TIMESTAMP_MAPPING;
+        public CompressedXContent getDefaultMapping(final IndexSettings indexSettings) {
+            final IndexSortConfig.FieldSortSpec[] fieldSortSpecs = indexSettings.getIndexSortConfig().sortSpecs;
+            if (fieldSortSpecs == null || fieldSortSpecs.length == 0) {
+                return DEFAULT_LOGS_TIMESTAMP_MAPPING_WITH_HOSTNAME;
+            }
+
+            for (final IndexSortConfig.FieldSortSpec fieldSortSpec : fieldSortSpecs) {
+                if (HOST_NAME.equals(fieldSortSpec.field)) {
```

Comment:
What if the user provides an override with "hostname" or "attributes.host.name"? 

I think we should just skip injecting anything other than timestamp when index sorting is explicitly set.

Reply:
I think I tested this right? Can you be more precise...what do you mean with override?

Reply:
An override is an explicit setting of index sorting. I'm trying to understand why this loop is needed, is it to pick the default setting from `IndexSortConfig`, as mentioned above?

Reply:
This method is used to merge default mappings with the result of template composition...which happens taking into account priorities and fields defined by our templates or custom templates. We **need** to provide them anyway because then whether they are used or not will depend on the composition logic.

Reply:
I will go for the original plan got the host.name field...we can't really detect in the getDefault method of IndexMode if the field was added to index sorting by us (injecting it in IndexSortConfig  or if it was the user explicitly adding it by means of index.sort.field .
In IndexSortConfig we already inject the host.name field if the sort config is empty...later on in getDefault we just make sure the field is mapped because otherwise sorting would not be possible

Reply:
I'm fine either way. But could we do something like this?

```java
if (indexSettings.getSettings().hasValue(IndexSortConfig.INDEX_SORT_FIELD_SETTING.getKey())) {
    return DEFAULT_LOGS_TIMESTAMP_MAPPING;
} else {
    return DEFAULT_LOGS_TIMESTAMP_MAPPING_WITH_HOSTNAME;
}
```

Reply:
Seems like that works. We don't actually inject an index setting in LogsDB, we just apply a default in `IndexSortConfig`.

Reply:
> I'm fine either way. But could we do something like this?
> 
> ```java
> if (indexSettings.getSettings().hasValue(IndexSortConfig.INDEX_SORT_FIELD_SETTING.getKey())) {
>     return DEFAULT_LOGS_TIMESTAMP_MAPPING;
> } else {
>     return DEFAULT_LOGS_TIMESTAMP_MAPPING_WITH_HOSTNAME;
> }
> ```

I think that would not work if we have something like

```
            settings:
              index:
                sort.field: [ ]
                sort.order: [ ]
                mode: logsdb
```

I imagine that this is kind of a corner case...but I would be careful just in case this is automatically generated in some way and an empty array is used for some reason.